### ISSUE TITLE
When using tf.keras.Model.predict(), the `batch_size` makes the results different

### ISSUE THREAD

**Issue by mengjiexu** [tensorflow/tensorflow#39663]
The code below is different:
images = np.array([cv2.imread('a.jpg'), cv2.imread('b.jpg')]) / 255.
model = tf.keras.applications.xception.Xception()
model.predict(images,  **batch_size**=1)

and:
model = tf.keras.applications.xception.Xception()
model.predict(images,  **batch_size**=2)

**Comment by mengjiexu** [comment]
I have change the BatchNormlization()(x, training=False)

**Comment by luozhouyang** [comment]
It seems that the inputs of the model are different, because `images` tensor is generated randomly.

**Comment by mengjiexu** [comment]
I have change the dscription, the images is same, the results is different

**Comment by ravikyram** [comment]
@mengjiexu 

Request you to share colab link or simple standalone code with supporting files to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by mengjiexu** [comment]
I mean with two same images(can be any images), same xception network, when predict with batch_size = 1 and batch_size=2, the result is different.

**Comment by ravikyram** [comment]
I have tried in colab with TF version 2.2.Please, find the gist [here.](https://colab.sandbox.google.com/gist/ravikyram/d4c498f25ce24fec03c5e9cab1aa8e0c/untitled914.ipynb)Is the the expected behavior?.Thanks!

**Comment by ravikyram** [comment]
@mengjiexu 

Any update on this issue please. Thanks!
